lib/vssh/libssh.c

```diff
@@ -1772,6 +1774,13 @@ static CURLcode myssh_statemach_act(struct Curl_easy *data, bool *block)
       /* during times we get here due to a broken transfer and then the
          sftp_handle might not have been taken down so make sure that is done
          before we proceed */
+      ssh_set_blocking(sshc->ssh_session, 0);
```

Comment:
This change is outside the version check. Is that intended and safe?

Reply:
Is intended, here is sftp shutdown phase, set session to nonblocking to make sure not stuck in sftp_close when network broken. 
It's safe, I have check the following steps SSH_SESSION_DISCONNECT/SSH_SESSION_FREE/SSH_QUIT there is no async api call.

Besides, the sftp download always use nonblocking session, and sftp or scp upload always use blocking session, so the sftp shutdown phase should be safe both in blocking and nonblocking.